src/uri.cr

```diff
@@ -144,6 +144,16 @@ class URI
     end
   end
 
+  # Returns `true` if URI has a *scheme* specified.
```

Comment:
I still think this is not enough. See https://github.com/crystal-lang/crystal/pull/6311#discussion_r199453079